Vizzuality/heco-invest
# Update search box and filters

## Description of the PR

This PR updates the search style and behaviour 

OBS: I had to refactor some components to avoid repeating code, so I changed some files that are not directly related to the task but are related to the search/filter functionality 

### Acceptance criteria

- The filter functionality in the search bar will be more prominent
DE decision: a “filter” button will be added to the search bar

-  When clicking the filter button, the search field, or the glass icon, the possible filters are displayed
DE decision: the already existing filter modal pops up below the search bar

- Filters related to the key words entered will be suggested automatically.
Note: functionality already in place; refined design to be applied in the FE

- When a filter is selected, there will be a way for the user to know the filter is active
DE decision: if users select a filter while searching, they will appear active meanwhile.

- When the search is triggered and the results displayed, there will be a way for the user to know which filters are applied
DE decision: when the results are ready, any filter applied during the search will appear as a tag below the search bar.

- Applies to search functionality in the homepage for consistency
Consider specificities of the homepage

## Testing instructions

1. **Discover Pages**

[mobile design](https://www.figma.com/file/KC49uMR9t1YARmFR0Sr6aN/HeCo---Work-in-progress?node-id=5087%3A87612)
[desktop design](https://www.figma.com/file/KC49uMR9t1YARmFR0Sr6aN/HeCo---Work-in-progress?node-id=7354%3A107568)
[behaviour](https://www.figma.com/proto/KC49uMR9t1YARmFR0Sr6aN/HeCo---Work-in-progress?page-id=2%3A4&node-id=7354%3A107563&scaling=scale-down&starting-point-node-id=6183%3A207104&show-proto-sidebar=1) 

- Go to the home page.
- You should be able to open the filters by clicking in any component of the search box.
- You should be able to select one filter from each type. If you click on a selected filter, it will be unselected. 
- The 'search' button on filters should apply all the selected filters.
- The 'clear filters' button should remove all the filters from the search (except the full text/search param)

- The active filters (if there are any) should be displayed as tags when the filters component is closed. 
- When clicking on an active filter tag, it will be removed from the search

- When typing, the suggestions will be displayed and the filters will be closed.
- When clicking on a tag, the filter should be applied. 

3. **Home page**

[mobile design](https://www.figma.com/file/KC49uMR9t1YARmFR0Sr6aN/HeCo---Work-in-progress?node-id=5093%3A111630)
[desktop design](https://www.figma.com/file/KC49uMR9t1YARmFR0Sr6aN/HeCo---Work-in-progress?node-id=2%3A3597)

Same as above +
- You should see the 'See full catalogue' button. 
- The 'See full catalogue' button should redirect you to the discover page.
- The 'search' button on filters should redirect you to the discover page with the correct filters.
- When clicking on a suggested filter tag, you should be redirected to the discover page with the filter applied. 

## Tracking
[1059](https://vizzuality.atlassian.net/browse/LET-1059)


https://user-images.githubusercontent.com/48164343/189907522-dbabfd8d-32ef-4fd7-a7d6-f0255c96bcb4.mov
